urwid/event_loop/asyncio_loop.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/python
+#
+# Urwid main loop code
+#    Copyright (C) 2004-2012  Ian Ward
+#    Copyright (C) 2008 Walter Mundt
+#    Copyright (C) 2009 Andrew Psaltis
+#
+#    This library is free software; you can redistribute it and/or
+#    modify it under the terms of the GNU Lesser General Public
+#    License as published by the Free Software Foundation; either
+#    version 2.1 of the License, or (at your option) any later version.
+#
+#    This library is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#    Lesser General Public License for more details.
+#
+#    You should have received a copy of the GNU Lesser General Public
+#    License along with this library; if not, write to the Free Software
+#    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# Urwid web site: https://urwid.org/
+
+
+from __future__ import annotations
+
+import asyncio
+import typing
+from collections.abc import Callable
+
+from .abstract_loop import EventLoop, ExitMainLoop
+
+__all__ = ("AsyncioEventLoop",)
+
+
+class AsyncioEventLoop(EventLoop):
+    """
+    Event loop based on the standard library ``asyncio`` module.
+
+    """
+    _we_started_event_loop = False
+
+    _idle_emulation_delay = 1.0/30  # a short time (in seconds)
```

Comment:
it still makes me sad that the asyncio loop needs to consume CPU while idle

Reply:
There's an old ticket with a discussion about how this could be fixed if you're interested

Reply:
I seen this PR, but need to invest extra time to restore PR and address comments.
This will be done as separate PR at short time.